experiments/modern/src/maki-interpreter/variable.js

```diff
@@ -16,10 +17,11 @@ class Variable {
     if (this._unsubscribeFromValue != null) {
       this._unsubscribeFromValue();
     }
-    if (this.typeName === "OBJECT") {
+    if (this.global && this.typeName === "OBJECT" && value && value.js_listenToAll) {
       this._unsubscribeFromValue = value.js_listenToAll((eventName, args) => {
         this._emitter.trigger(eventName, args);
       });
+      value.updateHooks(this, Object.keys(this._emitter._hooks));
```

Comment:
I don't think we need this. I'll comment elsewhere.

Reply:
So I'm almost sure this isn't the "right" way to do it, but I think it will be useful to be able to ask a `MakiObject`, "What hooks are the variables you are listening to currently hooked into?". This way we can only setup event bindings for hooks that are actually active.